examples/tinywl/LayerSurface.qml

```diff
@@ -0,0 +1,181 @@
+// Copyright (C) 2023 rewine <luhongxu@deepin.org>.
+// SPDX-License-Identifier: Apache-2.0 OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+
+import QtQuick
+import Waylib.Server
+
+LayerSurfaceItem {
+    id: layerSurface
+
+    required property WorkSpace workSpace
+    property int lastExclusiveZone: 0
+    property int lastExclusiveEdge: WaylandLayerSurface.AnchorNone
+    required property WaylandLayerSurface waylandSurface
+
+    surface: waylandSurface
+
+    z: zValformLayer(waylandSurface.layer)
+
+    OutputLayoutItem {
+        anchors.fill: parent
+        layout: QmlHelper.layout
+
+        onEnterOutput: function(output) {
+            waylandSurface.surface.enterOutput(output);
+        }
+        onLeaveOutput: function(output) {
+            waylandSurface.surface.leaveOutput(output);
+        }
+    }
+
+    function zValformLayer(layer) {
+        switch (layer) {
+          case WaylandLayerSurface.LayerBackground:
```

Comment:
Unless the enum value is not consistent with `wlr-layer-shell-unstable-v1(or wlroots)`, we need background(0)/bottom(1) under the xdg surface, which z value is 0

The conversion function  can move to the c++ part, so that we can use c++ enum value for the z value directly in qml